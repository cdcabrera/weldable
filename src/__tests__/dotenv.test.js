const path = require('path');
const crypto = require('crypto');
const dotenv = require('../dotenv');
const { contextPath } = require('../global');

describe('dotenv', () => {
  it('should return specific properties', () => {
    expect(dotenv).toMatchSnapshot('specific properties');
  });

  it('should attempt to load and instantiate the dotenv-webpack package', () => {
    expect(dotenv.setupWebpackDotenvFile()).toMatchSnapshot('setupWebpackDotenvFile');
  });

  it('should attempt to load dotenv files for webpack configuration', () => {
    const results = dotenv.setupWebpackDotenvFilesForEnv({ directory: contextPath, env: 'development' });

    expect(results.length).toBe(4);
    expect(
      results.map(params => {
        const { path: filePath, ...rest } = params.config;

        return { ...rest, path: path.basename(filePath) };
      })
    ).toMatchSnapshot('setupWebpackDotenvFilesForEnv');
  });

  it('should attempt to load dotenv files standalone and apply to process.env', () => {
    const dotenvParamName = 'DOTENV_TEST';
    const content = crypto.randomBytes(20).toString('hex');
    const { path: filePath, removeFixture } = generateFixture(`${dotenvParamName}=${content}`);

    dotenv.setupDotenvFile(filePath);
    expect(process.env[dotenvParamName]).toBe(content);
    removeFixture();
  });

  it('should attempt to load dotenv files as part of a working env and apply to process.env', () => {
    const dotenvParamName = 'DOTENV_TEST_ENV';
    const content = crypto.randomBytes(20).toString('hex');
    const { dir, removeFixture } = generateFixture(`${dotenvParamName}=${content}`, { filename: '.env.test', ext: '' });

    const result = dotenv.setupDotenvFilesForEnv({ env: 'test', relativePath: dir });

    expect(result[dotenvParamName]).toBe(content);
    removeFixture();
  });
});
