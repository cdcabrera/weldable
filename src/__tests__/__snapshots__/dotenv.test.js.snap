// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dotenv should attempt to load and instantiate the dotenv-webpack package: setupWebpackDotenvFile 1`] = `
Dotenv {
  "config": {
    "path": "./.env",
    "prefix": "process.env.",
    "silent": true,
    "systemvars": true,
  },
}
`;

exports[`dotenv should attempt to load dotenv files for webpack configuration: setupWebpackDotenvFilesForEnv 1`] = `
[
  {
    "path": ".env.development.local",
    "prefix": "process.env.",
    "silent": true,
    "systemvars": true,
  },
  {
    "path": ".env.development",
    "prefix": "process.env.",
    "silent": true,
    "systemvars": true,
  },
  {
    "path": ".env.local",
    "prefix": "process.env.",
    "silent": true,
    "systemvars": true,
  },
  {
    "path": ".env",
    "prefix": "process.env.",
    "silent": true,
    "systemvars": true,
  },
]
`;

exports[`dotenv should return specific properties: specific properties 1`] = `
{
  "dotenv": {
    "setupDotenvFilesForEnv": [Function],
    "setupWebpackDotenvFilesForEnv": [Function],
  },
  "setDotenvParam": [Function],
  "setupDotenvFile": [Function],
  "setupDotenvFilesForEnv": [Function],
  "setupWebpackDotenvFile": [Function],
  "setupWebpackDotenvFilesForEnv": [Function],
}
`;
